src/Compilers/Core/Portable/StrongName/CryptoBlobParser.cs

```diff
@@ -225,6 +233,17 @@ public unsafe static bool TryGetPublicKey(ImmutableArray<byte> blob, out Immutab
                 // The RSA magic key must match the blob id
                 if (type == PrivateKeyBlobId && magic == RSA2)
                 {
+                    RSAParameters privKey = default;
+                    privKey.Exponent = expBytes;
+                    privKey.Modulus = pubKeyData;
+                    privKey.P = blobReader.ReadBytes(modulusLength / 2);
```

Comment:
@bartonjs: Right now we are writing out to a SnPublicKey format where `type`, `version`, `algorithm id` are needed.  Do you have code that could produce this Sn key format so we can embed it in our output assemblies?